packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/model/inheritance/single-discriminator/src/Generated/Models/Eagle.cs

```diff
@@ -0,0 +1,27 @@
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace _Type.Model.Inheritance.SingleDiscriminator.Models
+{
+    public partial class Eagle : Bird
+    {
+        public Eagle(int wingspan) : base("eagle", wingspan) => throw null;
+
+        internal Eagle(IList<Bird> friends, IDictionary<string, Bird> hate, Bird partner, int wingspan, IDictionary<string, BinaryData> serializedAdditionalRawData) : base("eagle", wingspan, serializedAdditionalRawData) => throw null;
```

Comment:
we are hiding a parameter here in child type in a discriminated set in order to hard code the value of it.

But this kind of breaks our internal convention (if you remember) that when we want the "full constructor", we will compare the parameter count and requires it to be `count of properties plus 1`, here this convention is broken.

This is one of the reasons that I said this kind of convention is dangerous that ourselves could break it and cause weird issues.

We could make it simpler by just adding the `kind` and pass it through. The value of kind will always be `eagle` because we control all the invocation of this ctor, which ensures the value of it is always `eagle`.